File: Task 03/haskell/Subtask4/Task4.hs
import System.IO

printDiamond :: Int -> String
printDiamond n = unlines $ map generateLine linesList
  where
    linesList = [1..n] ++ [n-1, n-2..1]
    generateLine i = spaces ++ stars
      where
        spaces = replicate (n - i) ' '
        stars  = replicate (2 * i - 1) '*'

main :: IO ()
main = do
    inputHandle <- openFile "input.txt" ReadMode
    input <- hGetContents inputHandle
    let n = read input :: Int
    hClose inputHandle
    
    let diamond = printDiamond n
    
    outputHandle <- openFile "output.txt" WriteMode
    hPutStr outputHandle diamond
    hClose outputHandle

File: Task 03/C++/Subtask4/Task4.cpp
#include<iostream>
#include<fstream>
using namespace std;

int main()
{	
	int n;
	ifstream numfile("input.txt");
	numfile>>n;
	numfile.close();
	
	ofstream outnumfile("output.txt")
	
	
	for(int i = 1; i <= n; i++) {
		for(int j=1;j<n;j++){
			outnumfile<<" ";
		}
		
		for (int j=1;j<=(2*i-1);j++){
			outnumfile<<""*"";
		}
		outnumfile<<endl;
	}
	for(int i = n-1; i >= 1; i--) {
		for(int j = n; j > i; j--) {
			outnumfile<< " ";
		}
		for(int j = 1; j <= (2 * i - 1); j++) {
			outnumfile<<("*")
		}
		outnumfile<<endl;
		
	outnumfile.close()
	
	return 0;
}
